src/Symfony/Component/HtmlSanitizer/Parser/NativeParser.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HtmlSanitizer\Parser;
+
+/**
+ * Parser using PHP 8.4's new Dom API.
+ */
+final class NativeParser
+{
+    public function __construct()
+    {
+        if (\PHP_VERSION_ID < 80400) {
+            throw new \LogicException(self::class.' requires PHP 8.4 or higher.');
+        }
+    }
+
+    public function parse(string $html, string $context = 'body'): ?\Dom\Node
```

Comment:
DOM namespace classes should be imported. php-cs-fixer didn't detect this in fabbot, probably because it runs on PHP 8.3 and doesn't fix the code when the class doesn't exist.


```suggestion
    public function parse(string $html, string $context = 'body'): ?Node
```

Reply:
Actually we didn't enable this import rule. Usually we don't import symbols that don't follow the Vendor\Package\Etc convention, as is the case for most internal classes. They're short enough and that makes things easier to grasp in a snap.